app/helpers/generic_file_helper.rb

```diff
@@ -21,8 +21,9 @@ def render_download_link text = nil
     link_to (text || "Download"), download_path(@generic_file), { id: "file_download", target: "_new", data: { label: @generic_file.id } }
   end
 
-  def download_path gf
-    GenericFile.find(gf.id).doi_url || sufia.download_path(gf.id)
+  # sufia.download path is from ActionDispatch::Routing::PolymorphicRoutes#polymorphic_path
+  def download_path gf, options = {}
+    GenericFile.find(gf.id).doi_url || sufia.download_path(gf.id, options)
```

Comment:
Re: Routes

It is not related to polymorphic_path. It's coming from Rails by way of Sufia's routes file. Specifically [once Sufia defines a :downloads resource](https://github.com/projecthydra/sufia/blob/master/config/routes.rb#L36), Rails kicks into gear and [generates named route methods for it](https://github.com/rails/rails/blob/51c7ac142d31095d4c699f44cc44ddea627da1eb/actionpack/lib/action_dispatch/routing/route_set.rb#L279-L289). All rails route helpers are just defined as taking *args, an indistinct number of arguments. Which explains why the arity is -1, not -2 (which is super ambiguous since this was not written in C).

Doing it this way is mostly futureproofing. download_path(a, b={}) and download_path(a, *b) behave identically, until sufia or anyone else changes the download path route definition to add an extra param. If that happens, you'll have to change your download_path override, whereas it should(knock-on-wood, no guarantees) keep working in the latter definition.